src/v/cloud_storage/remote.cc

```diff
@@ -1069,12 +1069,7 @@ ss::future<upload_result> remote::delete_object(
             result = upload_result::failed;
             break;
         case cloud_storage_clients::error_outcome::key_not_found:
-            vassert(
-              false,
-              "Unexpected notfound outcome received when deleting object {} "
-              "from bucket {}",
-              path,
-              bucket);
+            result = upload_result::notfound;
```

Comment:
Looking at the usage of delete_objects and other methods, it seems that if we don't return success, the system could try to reschedule deletion. If there isn't a limit there, maybe it can be problematic (or at least annoying to read a failure in the log).

maybe we could just map the error to upload_result::success and log a WARN here

Reply:
Oh, I see now that the `s3_client` will simply return `no_response{}` in the case of a missing key. I agree, `abs_client` should do the same.